Products/ZenUtils/patches/pasmonkey.py

```diff
@@ -310,7 +318,17 @@ def updateCredentials(self, request, response, login, new_password):
     # user and return immediately if authenticated
     user_id, info = acl_users._tryEmergencyUserAuthentication(credentials)
 
-    if user_id is None:
+    request.SESSION.set('__blocked', None)
+    current = self.attempt.get(login, {})
```

Comment:
if you use `self.attempt.setdefault(login, {})` instead, you won't need to assign `current` back to `attempt[user_id]` further down in the code (line 367).